### PR TITLE
HPCC-22389 Admin group should allow fully qualified OU specification in config

### PR DESCRIPTION
**Pull request by RussWhitehead**
Currently, you can specify an LDAP group name in the configuration to
specify HPCC administrators. This group must be located under the "groups"
base dn which is also specified in the config. This PR allows you to
specify the group name as it does now, or a fully qualified OU that
resides elsewhere. With this new behaviour, the admin group membership
could be shared between HPCC deployments that do not share the same group
base DN.  This PR also fixes a misspelling the dali.xsd file

Signed-off-by: Russ Whitehead <william.whitehead@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [x] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by RussWhitehead** [comment]
@wangkx Please review

File: system/security/LdapSecurity/ldapconnection.cpp [hunk 1]
```diff
@@ -423,6 +418,36 @@ class CLdapConfig : implements ILdapConfig, public CInterface
...
+        }
+        else if (0 == stricmp("Directory Administrators", admin.str()))
+        {
+            admin.set("cn=Directory Administrators");//389 DirectoryServer
+        }
+        else if (nullptr == strstr(admin.str(), "CN=") && nullptr == strstr(admin.str(), "cn="))
+        {
+            //Group name only. Add group OU
+            StringBuffer sb;
+            sb.appendf("cn=%s,%s", admin.str(), group_basedn.str());
```
Comment (wangkx) [inline system/security/LdapSecurity/ldapconnection.cpp]: Could the admin be "Directory Administrators"?

File: system/security/LdapSecurity/ldapconnection.cpp [hunk 2]
```diff
@@ -423,6 +418,36 @@ class CLdapConfig : implements ILdapConfig, public CInterface
...
+            admin.set(sb);
+        }
+        else
+        {
+            //Fully qualified group name. No changes necessary
+        }
+        if (nullptr == strstr(admin.str(), "DC=") && nullptr == strstr(admin.str(), "dc="))
+            admin.appendf(",%s", m_basedn.str());//add DC (Domain Component)
+        LdapUtils::cleanupDn(admin, m_adminGroupDN);
+        DBGLOG("adminGroupName '%s'", m_adminGroupDN.str());
```
Comment (wangkx) [inline system/security/LdapSecurity/ldapconnection.cpp]: If this is for debug, it should be removed. If not, better use PROGLOG.

**Review by wangkx — COMMENTED** [review]
@RussWhitehead 2 comments

**Comment by RussWhitehead** [comment]
@wangkx Please rereview based on our discussion. I changed DBGLOG to PROGLOG and renamed the admin variable to adminGrp

**Review by wangkx — APPROVED** [review]
@RussWhitehead looks fine.

**Comment by RussWhitehead** [comment]
@richardkchapman Please merge

**Comment by richardkchapman** [comment]
@RussWhitehead What version should this target?

**Comment by RussWhitehead** [comment]
@richardkchapman 7.4.x I supposed, it's just an enhancement that nobody is asking for yet. Do I need to rebase?

**Comment by richardkchapman** [comment]
@RussWhitehead Please

**Comment by RussWhitehead** [comment]
Closed this PR and replaced with https://github.com/hpcc-systems/HPCC-Platform/pull/12726

**Pull request closed without merging**